polaris-for-vscode/server/src/server.ts

```diff
@@ -25,41 +25,83 @@ const connection = createConnection(ProposedFeatures.all);
 // Create a simple text document manager.
 const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
 
+type TokenGroup = {
+  [T in GroupedTokensKey]: RegExp;
+};
+
+type TokenArray = {
+  [key: string]: any;
+};
+
+allTokens as TokenArray;
+
+let tokenGroups: TokenGroup = {
+  color:
+    /color|background|shadow|border|column-rule|filter|opacity|outline|text-decoration/,
+  spacing: /margin|padding|gap|top|left|right|bottom/,
+  typography: /font|line-height/,
+  'z-index': /z-index/,
+  shape: /border/,
+  depth: /shadow/,
+  motion: /animation/,
+};
+
 connection.onInitialize((params: InitializeParams) => {
-	const capabilities = params.capabilities;
+  const capabilities = params.capabilities;
 
-	const result: InitializeResult = {
-		capabilities: {
-			textDocumentSync: TextDocumentSyncKind.Incremental,
-			// Tell the client that this server supports code completion.
-			completionProvider: {
+  const result: InitializeResult = {
+    capabilities: {
+      textDocumentSync: TextDocumentSyncKind.Incremental,
+      // Tell the client that this server supports code completion.
+      completionProvider: {
         triggerCharacters: ['--'],
-			}
-		}
-	};
+      },
+    },
+  };
 
-	return result;
+  return result;
 });
 
-
 // This handler provides the list of token completion items.
 connection.onCompletion(
-	(_textDocumentPosition: TextDocumentPositionParams): CompletionItem[] => {
-		// The pass parameter contains the position of the text document in
-		// which code complete got requested. For now we'll ignore but putting
-    // here for when we want to support context/position specific completions
+  (textDocumentPosition: TextDocumentPositionParams): CompletionItem[] => {
+    const doc = documents.get(textDocumentPosition.textDocument.uri);
+
+    // if the doc can't be found, return nothing
+    if (!doc) {
+      return [];
+    }
+
+    const currentText = doc.getText({
+      start: {line: textDocumentPosition.position.line, character: 0},
+      end: {line: textDocumentPosition.position.line, character: 1000},
```

Comment:
`1000` is probably safe but I if wonder we should leverage VSCode APIs to grab the actual end of line index. I don't know the exact APIs but I put together this rough example [referencing my VSCode extension](https://github.com/aaronccasanova/format-grid-areas/blob/12dc39d1836cd1ded6ab4ec02ad8064d17c2ce68/src/extension.ts#L18-L19):
```js
const currentText = doc.getText({
      start: {line: textDocumentPosition.position.line, character: 0},
      end: {
        line: textDocumentPosition.position.line,
        character: document.lineAt(textDocumentPosition.position).range.end.character,
      },
})
```

Reply:
yeah I wen with `1000` because the language server api (from what I could tell) does not have support for getting the end of the line position 😞 

There's probably a better, over-engineered approach that we can take, but for now this was working and I don't currently have a huge concern. This will at least unblock us for now and we can come back to it